### PR TITLE
Update v0.9.30

### PR DESCRIPTION
**Pull request by Marky-Shi**
#898  update deps to v0.9.30

### CODE REVIEW

File: runtime/litmus/src/tests/mod.rs [hunk 1]
```diff
@@ -24,19 +24,19 @@ pub mod setup {
...
 	runtime_common::run_transaction_payment_tests!();
 }
 
 mod xcm_parachain {
 	use crate::{
 		tests::setup::{
-			Call as RelayCall, Origin as RelayOrigin, ParaA, ParaB, Relay, RelayChainRuntime,
-			TestNet,
+			ParaA, ParaB, Relay, RelayChainRuntime, RuntimeCall as RelayCall,
+			RuntimeOrigin as RelayOrigin, TestNet,
```
Comment (wangminqi) [inline runtime/litmus/src/tests/mod.rs:36]: So these name changes should also apply to common/src/tests/setup/relays.rs  ,   right? But no change apply to these files yet. Need double check.

File: runtime/rococo/src/migration/P9100.rs [hunk 2]
```diff
@@ -31,7 +31,7 @@ where
 	<T as frame_system::Config>::Event: From<pallet_parachain_staking::Event<T>>,
 {
 	#[cfg(feature = "try-runtime")]
-	fn pre_upgrade() -> Result<(), &'static str> {
+	fn pre_upgrade() -> Result<Vec<u8>, &'static str> {
```
Comment (wangminqi) [inline runtime/rococo/src/migration/P9100.rs]: So these files are not used and only keep for history backup purposes. If no CI error, maybe we should just leave it?

I am open to any solution. minor issue.

**Review by wangminqi — CHANGES_REQUESTED** [review]

File: runtime/litentry/src/migration/P9100.rs [hunk 3]
```diff
@@ -31,7 +31,7 @@ where
 	<T as frame_system::Config>::Event: From<pallet_parachain_staking::Event<T>>,
 {
 	#[cfg(feature = "try-runtime")]
-	fn pre_upgrade() -> Result<(), &'static str> {
```
Comment (wangminqi) [inline runtime/litentry/src/migration/P9100.rs]: same above

File: runtime/litmus/src/tests/mod.rs (hunk 1 above)
Comment (Marky-Shi) [reply runtime/litmus/src/tests/mod.rs:36]: I compared the moonbeam's code, which is how they implemented it in version 0.9.30
image

Comment (wangminqi) [reply runtime/litmus/src/tests/mod.rs:36]: check. Looks okay.

**Review by wangminqi — APPROVED** [review]

File: .github/workflows/build-and-test.yml [hunk 4]
```diff
@@ -107,6 +107,12 @@ jobs:
           target: wasm32-unknown-unknown
           default: true
 
+      - name: Install protobuf
+        run: |
+          sudo apt-get update && \
+          sudo apt-get upgrade -y && \
+          sudo apt-get install -yq openssl clang libclang-dev cmake  protobuf-compiler
```
Comment (Kailai-Wang) [inline .github/workflows/build-and-test.yml:114]: Can you please remove the `upgrade -y` line and only install the needed deps? The step name is "install protobuf" but you are installing a lot more than just protobuf

File: .github/workflows/build-and-test.yml [hunk 5]
```diff
@@ -299,6 +305,12 @@ jobs:
           chmod +x $HOME/.local/bin/sccache
           echo "$HOME/.local/bin" >> $GITHUB_PATH
 
+      - name: Install protobuf
+        run: |
+         sudo apt-get update && \
+         sudo apt-get upgrade -y && \
+         sudo apt-get install -yq openssl clang libclang-dev cmake  protobuf-compiler
```
Comment (Kailai-Wang) [inline .github/workflows/build-and-test.yml:312]: Same as above

File: runtime/litentry/src/weights/pallet_proxy.rs [hunk 6]
```diff
@@ -138,20 +138,18 @@ impl<T: frame_system::Config> pallet_proxy::WeightInfo for WeightInfo<T> {
...
 	// Storage: Proxy Proxies (r:1 w:1)
-	/// The range of component `p` is `[1, 31]`.
-	fn anonymous(p: u32, ) -> Weight {
-		Weight::from_ref_time(35_206_000 as u64)
-			// Standard Error: 1_000
-			.saturating_add(Weight::from_ref_time(61_000 as u64).saturating_mul(p as u64))
+	fn create_pure(p: u32, ) -> Weight {
+		Weight::from_ref_time(31_077_000 as u64)
+			// Standard Error: 3_000
+			.saturating_add(Weight::from_ref_time(37_000 as u64).saturating_mul(p as u64))
```
Comment (Kailai-Wang) [inline runtime/litentry/src/weights/pallet_proxy.rs:144]: I see in the runtime setting `()` is used so it probably won't have an impact. But still, this doesn't look like automatically generated

**Review by Kailai-Wang — APPROVED** [review]
Principally looks good, only a few small things. Thanks

File: .github/workflows/build-and-test.yml (hunk 4 above)
Comment (Marky-Shi) [reply .github/workflows/build-and-test.yml:114]: Thanks for pointing out the deficiencies in the code, for this issue I will open an issue specifically to fix it

File: .github/workflows/build-and-test.yml (hunk 5 above)
Comment (Marky-Shi) [reply .github/workflows/build-and-test.yml:312]: Same as above

File: runtime/litentry/src/weights/pallet_proxy.rs (hunk 6 above)
Comment (Marky-Shi) [reply runtime/litentry/src/weights/pallet_proxy.rs:144]: It is to ensure that the compilation passes, so it is copied from the substrate,the weight file of pallet-proxy I will put it in #887  to solve together.

**Pull request merged**